content/react-part-2/reading/conditional-rendering/index.md

```diff
@@ -0,0 +1,452 @@
+   function RainOrShine(props){
+      const isRainy = props.isRainy;
+      const testWord = props.testWord;
+      if((isRainy) && (testWord === "thunder")) {
+         return(
+            <Rainy />
+         );
+      }
+   }
+   ```
+{{% /notice %}}
+
+
+
+We would need to add the `testWord` to the `App` function.
+
+{{% notice blue "Example: Updating App()" "rocket" %}}
+```react{linenos=table,hl_lines=[14],linenostart=1}
+   export default function App(){
+      return(
+         <div>
+         <RainOrShine 
+            isRainy={true}
+            testWord={"thunder"}
+         />
+         </div>
+      );
+   }
+   ```
+{{% /notice %}}
+
+{{% expand "Output" %}}
+   ## Bring your umbrella!
+{{% /expand %}}
+
+
+If `testWord` was not `"thunder"`, then nothing would render.
+
+{{% notice blue "Example: Updating App()" "rocket" %}}
+   ```react{linenos=table,hl_lines=[14],linenostart=1}
+   export default function App(){
+      return(
+      <div>
+         <RainOrShine 
+            isRainy={true}
+            testWord={"kangaroo"}
+         />
+      </div>
+      );
+   }
+   ```
+{{% /notice %}}
+
+{{% expand "Output" %}}
+   ##  
+
+   (Nothing is rendered.  This output represents a blank browser)
+{{% /expand %}}
+
+{{% notice blue "Note" "rocket" %}}
+   You could also try using the logical Or `||` and the logical not `!` in your application too.
+
+   These will work, but are not as useful as the `&&`.
+{{% /notice %}}
+
+
+## Ternary Operator
+
+The **ternary operator** is a single-line expression of an `if/else` statement.  
+
+### Syntax
+{{% notice blue "Basic Syntax" "rocket" %}}
+   ```js{linenos=table,hl_lines=[],linenostart=1}
+   condition ? ifTrueExpression : ifFalseExpression
+   ```
+{{% /notice %}}
+
+This would look like the following in JavaScript:
+{{% notice blue "Example Ternary Syntax" "rocket" %}}
+   ```js{linenos=table,hl_lines=[],linenostart=1}
+   let creature = "cat";
+   let creatureSound = creature = "cat" ? "Meow!" : "Woof!";  
+   console.log(creatureSound)
+   ```
+{{% /notice %}}
+
+{{% expand "Output" %}}
+   Meow! 
+{{% /expand %}}
+
+Ternary statements are simplified `if/else` blocks.  We could update our `RainOrShine` function with a ternary operator like this:
+
+{{% notice blue "Example: RainOrShine as Ternary" "rocket" %}}
+   ```react{linenos=table,hl_lines=[1,11],linenostart=1}
+    function RainOrShine(props){
+      const isRainy = props.isRainy;
+      return isRainy ?  <Rainy /> : <Sunny />
+   }
+
+
+   function App() {
+      return (
+         <div>
+            <RainOrShine
+            isRainy={true}
+            />
+         </div>
+      });
+   }
+   ```
+{{% /notice %}}
+
+{{% expand "Output" %}}
+   ## Bring your umbrella!
+{{% /expand %}}
+
+
+## Other applications for Conditional Rendering
+
+Conditional Rendering can also be applied to the styling of elements. 
+Let's add some to our `RainOrShine` function.  We will need
```

Comment:
Question ?  We will need What??  Found this confusing.

Reply:
well, that was a cliffhanger for sure!  added more information to make this less suspensful.

Reply:
(line 298 - not sure if this comment is lined up correctly right now)